core/src/test/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/TestWorkflow.java

```diff
@@ -132,22 +162,26 @@ public WorkflowStepPrompt getPrompt() {
 
     @Override
     public String getLogMessage() {
-      return String.format("Processed integer %d", progressStep.getInput());
+      return String.format("Processed integer %d", integerProgressStep.getInput());
     }
 
     @Override
     public ProgressStep getProgressStep() {
-      return progressStep;
+      return integerProgressStep;
+    }
+
+    public IntegerProgressStep getIntegerProgressStep() {
```

Comment:
This method seems very similar to the overridden method `getProgressStep` above, but `getProgressStep` has to return a `ProgressStep` to be used in `AbstractWorkflow`.  Also, we want to get the concrete type out of an `IntegerWorkflowStep` so that we can use its specific methods.

A possible solution would be to have:
```
<T extends ProgressStep> T getProgressStep();
```
but this would require a cast and users to know the type `T` in advance.  It also makes chaining these calls more difficult as well.

Both solutions appear to have drawbacks.

Reply:
Would it ever be useful to get the `IntegerProgressStep`? Maybe use a method that just returns the `Integer` value instead, since that's what you're really after?